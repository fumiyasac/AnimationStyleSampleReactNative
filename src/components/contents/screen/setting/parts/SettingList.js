/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 * @flow
 */

import React, { Component } from 'react';

import {
  Content,
  List
} from 'native-base';

import SettingListIntro from './SettingListIntro';
import SettingListItem from './SettingListItem';

export default class SettingList extends React.Component {
  render() {
    const { settingItems } = this.props;
    return (
      <Content>
        <SettingListIntro />
        <List dataArray={settingItems.items} renderRow={ (item) =>
          <SettingListItem title={item.title} onPressListItem={item.onPressListItem} />
        } />
      </Content>
    );
  };
}
