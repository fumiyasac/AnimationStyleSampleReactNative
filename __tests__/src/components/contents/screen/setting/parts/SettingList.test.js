import 'react-native';
import React from 'react';
import SettingList from '../../../../../../../src/components/contents/screen/setting/parts/SettingList';

import renderer from 'react-test-renderer';

it('SettingListコンポーネントが正しく表示されること', () => {

  const onPressListItemMock = jest.fn();

	let stubObject = {
		"items": [
			{ title: "設定コンテンツその1", onPressListItem: onPressListItemMock() },
			{ title: "設定コンテンツその2", onPressListItem: onPressListItemMock() },
			{ title: "設定コンテンツその3", onPressListItem: onPressListItemMock() },
		]
	};

  const tree = renderer.create(
		<SettingList settingItems={stubObject} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});