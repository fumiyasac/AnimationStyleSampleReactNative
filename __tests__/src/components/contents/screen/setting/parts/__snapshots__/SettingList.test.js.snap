// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SettingListコンポーネントが正しく表示されること 1`] = `
<RCTScrollView
  automaticallyAdjustContentInsets={false}
  contentContainerStyle={
    Array [
      Object {
        "padding": undefined,
      },
      undefined,
    ]
  }
  contentInset={
    Object {
      "bottom": 0,
    }
  }
  enableAutomaticScroll={true}
  enableOnAndroid={false}
  enableResetScrollToCoords={true}
  extraHeight={75}
  extraScrollHeight={0}
  getScrollResponder={[Function]}
  handleOnScroll={[Function]}
  keyboardDismissMode="interactive"
  keyboardOpeningTime={250}
  keyboardShouldPersistTaps="handled"
  keyboardSpace={0}
  onScroll={[Function]}
  resetKeyboardSpace={[Function]}
  resetScrollToCoords={
    Object {
      "x": 0,
      "y": 0,
    }
  }
  scrollEventThrottle={1}
  scrollForExtraHeightOnAndroid={[Function]}
  scrollIntoView={[Function]}
  scrollToEnd={[Function]}
  scrollToFocusedInput={[Function]}
  scrollToPosition={[Function]}
  showsVerticalScrollIndicator={true}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
        "flex": 1,
      },
      Object {
        "backgroundColor": "#ffffff",
        "flex": 1,
      },
    ]
  }
  update={[Function]}
  viewIsInsideTabBar={false}
  virtual={undefined}
>
  <View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={null}
    >
      <View
        avatar={true}
        noBorder={true}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "borderBottomWidth": null,
            "borderColor": "#c9c9c9",
            "flexDirection": "row",
            "marginLeft": 16,
            "paddingBottom": null,
            "paddingRight": null,
            "paddingTop": null,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "flex-start",
              "alignSelf": "flex-start",
              "flex": 0,
              "flexDirection": "row",
              "paddingTop": 14,
            }
          }
        >
          <Image
            source={
              Object {
                "testUri": "../../../src/components/contents/screen/setting/assets/avatar_fumiyasac.jpg",
              }
            }
            style={
              Object {
                "borderRadius": 28,
                "height": 56,
                "width": 56,
              }
            }
          />
        </View>
        <View
          style={
            Object {
              "alignItems": null,
              "alignSelf": null,
              "borderBottomWidth": null,
              "borderColor": "#c9c9c9",
              "flex": 1,
              "marginLeft": 15,
              "paddingBottom": 10,
              "paddingTop": 10,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#000",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "marginLeft": null,
                  "marginRight": 10,
                },
                Object {
                  "fontSize": 14,
                  "fontWeight": "bold",
                  "paddingTop": 6,
                },
              ]
            }
            uppercase={false}
          >
            作者: fumiyasac
          </Text>
          <Text
            note={true}
            style={
              Array [
                Object {
                  "color": "#808080",
                  "fontFamily": "System",
                  "fontSize": 14,
                  "fontWeight": "200",
                  "marginLeft": null,
                  "marginRight": 10,
                },
                Object {
                  "fontSize": 13,
                  "paddingBottom": 8,
                  "paddingTop": 8,
                },
              ]
            }
            uppercase={false}
          >
            これまではデザイナー → サーバーサイドエンジニア → iOSやReactNative等をするようになったこともあって、UI周りの実装に関するTIPS等を僭越ながら取り組んだり登壇したりしています。
          </Text>
        </View>
      </View>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={null}
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "borderBottomWidth": 0.5,
              "borderColor": "#c9c9c9",
              "flexDirection": "row",
              "marginLeft": 16,
              "paddingBottom": 13,
              "paddingRight": 16,
              "paddingTop": 13,
            },
            Object {
              "height": 50,
            },
          ]
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": undefined,
                "fontSize": 12,
              },
              Array [
                Object {
                  "color": "#000",
                  "fontSize": 30,
                },
                Object {
                  "color": "#000",
                  "fontSize": 30,
                },
                Object {
                  "fontSize": 16,
                },
                Object {
                  "color": "#00aced",
                },
              ],
              Object {
                "fontFamily": "Ionicons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
        <Text
          style={
            Array [
              Object {
                "alignSelf": "center",
                "color": "#000",
                "fontFamily": "System",
                "fontSize": 16,
              },
              Object {
                "color": "#555555",
                "fontSize": 13,
                "fontWeight": "normal",
                "paddingLeft": 8,
              },
            ]
          }
          uppercase={false}
        >
          Twitter
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={null}
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "borderBottomWidth": 0.5,
              "borderColor": "#c9c9c9",
              "flexDirection": "row",
              "marginLeft": 16,
              "paddingBottom": 13,
              "paddingRight": 16,
              "paddingTop": 13,
            },
            Object {
              "height": 50,
            },
          ]
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": undefined,
                "fontSize": 12,
              },
              Array [
                Object {
                  "color": "#000",
                  "fontSize": 30,
                },
                Object {
                  "color": "#000",
                  "fontSize": 30,
                },
                Object {
                  "fontSize": 16,
                },
                Object {
                  "color": "#3b5998",
                },
              ],
              Object {
                "fontFamily": "Ionicons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
        <Text
          style={
            Array [
              Object {
                "alignSelf": "center",
                "color": "#000",
                "fontFamily": "System",
                "fontSize": 16,
              },
              Object {
                "color": "#555555",
                "fontSize": 13,
                "fontWeight": "normal",
                "paddingLeft": 8,
              },
            ]
          }
          uppercase={false}
        >
          Facebook
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={null}
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "borderBottomWidth": 0.5,
              "borderColor": "#c9c9c9",
              "flexDirection": "row",
              "marginLeft": 16,
              "paddingBottom": 13,
              "paddingRight": 16,
              "paddingTop": 13,
            },
            Object {
              "height": 50,
            },
          ]
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": undefined,
                "fontSize": 12,
              },
              Array [
                Object {
                  "color": "#000",
                  "fontSize": 30,
                },
                Object {
                  "color": "#000",
                  "fontSize": 30,
                },
                Object {
                  "fontSize": 16,
                },
                Object {
                  "color": "#333333",
                },
              ],
              Object {
                "fontFamily": "Ionicons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
        <Text
          style={
            Array [
              Object {
                "alignSelf": "center",
                "color": "#000",
                "fontFamily": "System",
                "fontSize": 16,
              },
              Object {
                "color": "#555555",
                "fontSize": 13,
                "fontWeight": "normal",
                "paddingLeft": 8,
              },
            ]
          }
          uppercase={false}
        >
          Github
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={null}
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "borderBottomWidth": 0.5,
              "borderColor": "#c9c9c9",
              "flexDirection": "row",
              "marginLeft": 16,
              "paddingBottom": 13,
              "paddingRight": 16,
              "paddingTop": 13,
            },
            Object {
              "height": 50,
            },
          ]
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": undefined,
                "fontSize": 12,
              },
              Array [
                Object {
                  "color": "#000",
                  "fontSize": 30,
                },
                Object {
                  "color": "#000",
                  "fontSize": 30,
                },
                Object {
                  "fontSize": 16,
                },
                Object {
                  "color": "#0077b5",
                },
              ],
              Object {
                "fontFamily": "Ionicons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
        <Text
          style={
            Array [
              Object {
                "alignSelf": "center",
                "color": "#000",
                "fontFamily": "System",
                "fontSize": 16,
              },
              Object {
                "color": "#555555",
                "fontSize": 13,
                "fontWeight": "normal",
                "paddingLeft": 8,
              },
            ]
          }
          uppercase={false}
        >
          Slideshare
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={null}
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "borderBottomWidth": 0.5,
              "borderColor": "#c9c9c9",
              "flexDirection": "row",
              "marginLeft": 16,
              "paddingBottom": 13,
              "paddingRight": 16,
              "paddingTop": 13,
            },
            Object {
              "height": 50,
            },
          ]
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": undefined,
                "fontSize": 12,
              },
              Array [
                Object {
                  "color": "#000",
                  "fontSize": 30,
                },
                Object {
                  "color": "#000",
                  "fontSize": 30,
                },
                Object {
                  "fontSize": 16,
                },
                Object {
                  "color": "#55c500",
                },
              ],
              Object {
                "fontFamily": "Ionicons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
        <Text
          style={
            Array [
              Object {
                "alignSelf": "center",
                "color": "#000",
                "fontFamily": "System",
                "fontSize": 16,
              },
              Object {
                "color": "#555555",
                "fontSize": 13,
                "fontWeight": "normal",
                "paddingLeft": 8,
              },
            ]
          }
          uppercase={false}
        >
          Qiita
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
